src/modeler/AutoRest.Swagger/Validation/ListOperationNamingWarning.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+using System.Linq;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class ListOperationNamingWarning : TypedRule<Dictionary<string,Dictionary<string, Operation>>>
+    {
+        private readonly string XmsPageable = "x-ms-pageable";
+        private readonly Regex regex = new Regex(@".+_List(.*)$", RegexOptions.IgnoreCase);
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string,Dictionary<string, Operation>> entity, RuleContext context)
+        {
+            // get all operation objects that are either of get or post type
+            var serviceDefinition = (ServiceDefinition)context.Root;
+            var listingOperations = entity.Values.SelectMany(opDict=>opDict.Where(pair => pair.Key.Equals("get") || pair.Key.Equals("post")));
+            foreach (var opPair in listingOperations)
+            {
+                // if the operation id is already of the type _list we can skip this check
+                if (regex.IsMatch(opPair.Value.OperationId))
+                { continue; }
+
+                if(AutoRest.Swagger.Model.Utilities.ValidationUtilities.IsXmsPageableOrArrayResponseOperation(opPair.Value, serviceDefinition))
```

Comment:
seeing this I wonder why the `ValidationUtilities` are not under `AutoRest.Swagger.Model.Utilities.ValidationUtilities`